MaxText/max_utils.py

```diff
@@ -115,6 +119,23 @@ def write_metrics_for_gcs(metrics, step, config, running_metrics):
     running_metrics = [] # reset running_metrics to empty list
   return running_metrics
 
+def write_train_config_for_gcs(config):
```

Comment:
lol this is hilarious and pragmatic. but can we just copy.deepcopy before user_init right?

Reply:
Haha yeah it is. We could upload the config [here](https://github.com/google/maxtext/blob/a54a940d3a2b7bf2851289ef8b7335ffbeacb751/MaxText/pyconfig.py#L174) before the `_list_to_tuples` conversion happens so that we don't have to convert back however then the config that gets uploaded would not have the initialized values like `emb_dim`, `num_query_heads`, etc.

Reply:
Sounds good, I changed it to upload the config first as `base_config.yml` and then again as `initialized_config.yml` after the values get initialized.

Reply:
Actually, I changed it to just upload once in pyconfig after the values are initialized but before the type conversion. So now the config we upload will have all of the values and we won't have to do a `_tuples_to_lists`.